### DISCUSSION TITLE
HACK: how to load pandas dataframe in aggrid

### DISCUSSION THREAD

**Discussion by daniele-liprandi** [zauberzeug/nicegui#521] [Show and tell]
Hi,
by using justpy.AgGrid I was able to obtain an aggrid dictionary from a pandas dataframe and represent it in aggrid. Here is the code, which I hope could be useful for you guys to implement it directly

```py
from justpy import AgGrid

def options_dict_from_df():
    justpygrid = AgGrid()
    justpygrid .load_pandas_frame(pd.read_csv("test.csv"))
    return {'columnDefs':grid.convert_object_to_dict()["def"]["columnDefs"],
           'rowData':grid.convert_object_to_dict()["def"]["rowData"],
           'rowSelection': 'multiple',}   

grid= ui.aggrid(getoptionsforaggrid())

def exupdate():
    grid.options['rowData'] = getoptionsforaggrid()['rowData']
    grid.update()
```

I noted that `grid.options['rowData'] = getoptionsforaggrid()['rowData']` works, while `grid.options = getoptionsforaggrid()`
I also noted that I cannot visualize the aggrid inside a card, as it automatically collapse to width 0.

Sorry for the current lack of further details, feel free to ask me more questions if needed

**Comment by falkoschindler** [comment]
Thanks for sharing, @TheFaither!
This seems to be related to https://github.com/zauberzeug/nicegui/discussions/30. Maybe this is a way to do it without JustPy?

**Comment by abulka** [comment]
> I cannot visualize the aggrid inside a card, as it automatically collapse to width 0.

If you put some other elements in the card, then the card widens and you can see the aggrid. Or just style the card to be forced wider:

```python
from nicegui import ui


with ui.card().classes('w-2/3 items-center'):
# with ui.card():
    # ui.label('Card content')
    # ui.button('Add label', on_click=lambda: ui.label('Click!'))
    grid = ui.aggrid({
        'columnDefs': [
            {'headerName': 'Name', 'field': 'name'},
            {'headerName': 'Age', 'field': 'age'},
        ],
        'rowData': [
            {'name': 'Alice', 'age': 18},
            {'name': 'Bob', 'age': 21},
            {'name': 'Carol', 'age': 42},
        ],
        'rowSelection': 'multiple',
    }).classes('max-h-40')

ui.run()
```

**Comment by falkoschindler** [comment]
I'll close this discussion since creating AG Grids from pandas dataframes is now part of the official NiceGUI API:
https://nicegui.io/documentation/aggrid#create_grid_from_pandas_dataframe
